### DISCUSSION TITLE
[Dialog] how do I render dialog close to the element where the trigger is ?

### DISCUSSION THREAD

**Discussion by yuvigrg** [radix-ui/primitives#1875] [Help] [answered]
Hi,

I am following the pattern here where I have the dialog to appear inside a containing div. The example for [Custom portal container](https://www.radix-ui.com/docs/primitives/components/dialog#custom-portal-container) works and dialog is contained inside the parent component. I would like to add another effect like this i.e. display where the trigger was.  

![image](https://user-images.githubusercontent.com/57096255/211034847-9cd4ed49-678f-4bfb-beb2-9a5a7700bdd1.png)

Is there a native functionality ?

**Comment by benoitgrelard** [comment, marked as answer]
Hey @yuvigrg,

You are talking about `Dialog` but the screenshot shows `DropdownMenu` so it's a little confusing.
However, if what you are after is for the content not to be portalled away from its natural location in the DOM, simply don't use the `Portal` part and it will be rendered right next to the trigger in the DOM.

However, that will have an impact on visual layering. You may come across z-index issues that are difficult to get out of.
But otherwise that's completely possible this way.

**Reply by yuvigrg** [reply]
@benoitgrelard i am sorry about the confusion. What I am after is having the dialog next to the trigger button hence i showed the example of the dropdown. A better example might be this one where `add from url` is a trigger button and the dialog appear just underneath. 

With the custom portal, the portal appear within the div but covers the trigger button. I have tried without the suggested method of not using `Portal` and the result was the same. 

![image](https://user-images.githubusercontent.com/57096255/211043193-1e7c0cb4-84fe-48d7-9b3f-7cf243d3f5d3.png)

PS. it might be a react problem to solve but thought i would ask before investigating it further.

**Reply by benoitgrelard** [reply]
Are you after `Popover` rather than `Dialog` then?

**Reply by yuvigrg** [reply]
ah looks like, Popover is a better fit than Dialog. Thank you for pointing it out to me. I will give it a try now
